mewarpx/mewarpx/mwxrun.py

```diff
@@ -135,12 +165,150 @@ def init_grid(self, lower_bound, upper_bound, number_of_cells, **kwargs):
             upper_boundary_conditions=boundary_conditions['fields']['max'],
             lower_boundary_conditions_particles=boundary_conditions['particles']['min'],
             upper_boundary_conditions_particles=boundary_conditions['particles']['max'],
-            warpx_max_grid_size=max(number_of_cells)//kwargs.get('min_tiles', 1)
         )
 
         self._set_grid_params()
         self._print_grid_params()
 
+        # appropriately calculate the minimum number of tiles parameter
+        min_tiles = kwargs.get('min_tiles', None)
+        if min_tiles is not None:
+            if self.dim == 1:
+                self.grid.max_grid_size = self.nz // min_tiles
+            elif self.dim == 2:
+                max_nx_nz = max(self.nx, self.nz)
+                if min_tiles <= max(self.nx/self.nz, self.nz/self.nx):
+                    # All the tiles will be sliced in one dimension only
+                    self.grid.max_grid_size = max_nx_nz // min_tiles
+                else:
+                    # MIN_TILES is the smallest factor of 2 that will give at
+                    # least MIN_TOTAL_TILES total number of grids
+                    MAX_AREA_PER_TILE = np.sqrt(self.nx*self.nz/min_tiles)
+                    self.grid.max_grid_size = (
+                        max_nx_nz
+                        // int(2. * np.ceil(max_nx_nz / MAX_AREA_PER_TILE / 2.))
+                    )
```

Comment:
```suggestion
                    # max_grid_size is the smallest factor of 2 that will give at
                    # least min_tiles total number of grids
                    max_length_per_tile = np.sqrt(self.nx*self.nz/min_tiles)
                    self.grid.max_grid_size = (
                        max_nx_nz
                        // int(2. * np.ceil(max_nx_nz / max_length_per_tile / 2.))
                    )
```

Reply:
What fool called a linear dimension `area` anyway?

(Note for posterity: Yes, it was me)